MathGame.VincentyArmstrong/MathGame.VincentyArmstrong/UserInterface.cs

```diff
@@ -0,0 +1,125 @@
+﻿namespace MathGame.VincentyArmstrong;
+
+public class UserInterface
+{
+    private int questions;
+    private bool hardMode;
+    private List<Game> games = new();
+    private bool quit;
+
+    public void Run()
+    {
+        while (!quit)
+        {
+            MainMenuDisplay();
+            MainMenuInput();
+        }
+    }
+
+    private static void MainMenuDisplay()
+    {
+        Console.Clear();
+        Console.WriteLine(
+        @"
+            ---------------------------------------------
+            What game would you like to play today? Choose from the options below:
+            A - Addition
+            S - Subtraction
+            M - Multiplication
+            D - Division
+            H - View History
+            Q - Quit the program
+            ---------------------------------------------");
+    }
+   
+    private void MainMenuInput()
+    {
+        switch (Console.ReadLine().Trim().ToLower())
+        {
+            case "a":
+                Setup(GameType.Addition);
+                break;
+            case "s":
+                Setup(GameType.Subtraction);
+                break;
+            case "m":
+                Setup(GameType.Multiplication);
+                break;
+            case "d":
+                Setup(GameType.Division);
+                break;
+            case "h":
+                ViewHistory();
+                break;
+            case "q":
+                Exit();
+                break;
+            default:
+                break;
+        }
+    }
+
+    private void Setup(GameType gameType)
+    {
+        Questions();
+        Difficulty();
+        Game game = new Game(gameType, questions, hardMode);
+        games.Add(game);
+    }
+
+    private void Questions()
+    {
+        Console.Clear();
+        Console.WriteLine("How many questions do you wish to be asked?");
+
+        var input = Console.ReadLine();
+        while (!int.TryParse(input, out questions) || questions <= 0)
+        {
+            Console.WriteLine("Please enter a whole number larger than zero");
+            input = Console.ReadLine();
+        }
+    }
+
+    private void Difficulty()
```

Comment:
Same as the above method, I would suggest a more detailed name so that it is clear what the method does.

```suggestion
    private void GetDifficultyFromUser()
```